.github/actions/wait-for-check-completion/src/index.js

```diff
@@ -0,0 +1,194 @@
+        throw new Error(`Timeout: Checks [${pendingChecksList}] did not complete within ${timeout} seconds`);
+    }
+    return checkResults;
+}
+
+/**
+ * Main function
+ */
+async function run() {
+    try {
+        // Get inputs
+        const ref = core.getInput('ref', { required: true });
+        const checkNames = core.getInput('check-names', { required: true });
+        const token = core.getInput('repo-token', { required: true });
+        const waitInterval = parseInt(core.getInput('wait-interval') || '10', 10);
+        const timeout = parseInt(core.getInput('timeout') || '600', 10);
+
+        // Get repository info (default to current repository)
+        const owner = core.getInput('owner') || github.context.repo.owner;
+        const repo = core.getInput('repo') || github.context.repo.repo;
+
+        // Parse check names
+        const checkNamesArray = checkNames.split(',').map(name => name.trim()).filter(name => name.length > 0);
+
+        if (checkNamesArray.length === 0) {
+            throw new Error('No valid check names provided');
+        }
+
+        core.info(`Repository: ${owner}/${repo}`);
+        core.info(`Reference: ${ref}`);
+        core.info(`Check names: [${checkNamesArray.join(', ')}]`);
+
+        // Validate inputs
+        if (waitInterval < 1 || waitInterval > 300) {
+            throw new Error('wait-interval must be between 1 and 300 seconds');
+        }
+
+        if (timeout < 1 || timeout > 3600) {
+            throw new Error('timeout must be between 1 and 3600 seconds');
+        }
+
+        // Create GitHub API client
+        const octokit = github.getOctokit(token);
+
+        // Wait for the checks to complete
+        const results = await waitForChecks(octokit, owner, repo, ref, checkNamesArray, waitInterval, timeout);
+
+        // Analyze overall results
+        const allConclusions = Object.values(results).map(r => r.conclusion);
+        const allStatuses = Object.values(results).map(r => r.status);
+
+        // Determine overall conclusion
+        let overallConclusion;
+        if (allConclusions.every(c => c === 'success')) {
+            overallConclusion = 'success';
+        } else if (allConclusions.some(c => ['failure', 'cancelled', 'timed_out'].includes(c))) {
+            overallConclusion = 'failure';
+        } else if (allConclusions.some(c => c === 'action_required')) {
+            overallConclusion = 'action_required';
+        } else {
+            overallConclusion = 'mixed';
+        }
+
+        // Set outputs
+        core.setOutput('status', allStatuses.every(s => s === 'completed') ? 'completed' : 'mixed');
+        core.setOutput('conclusion', overallConclusion);
+        core.setOutput('results', JSON.stringify(results));
+
+        // Log results
+        for (const [checkName, result] of Object.entries(results)) {
+            core.info(`${checkName}: ${result.status} (${result.conclusion})`);
```

Comment:
Both status and conclusion could be `mixed`, it might be not clear in logs which goes first, let's add explicit labels
```suggestion
            core.info(`${checkName}: Status "${result.status}", Conclusion "${result.conclusion}"`);
```